ci-automation/vendor-testing/equinix_metal.sh

```diff
@@ -0,0 +1,118 @@
+# required for get_git_channel
+source sdk_lib/sdk_container_common.sh
+
+channel="$(get_git_channel)"
+
+mkdir -p "${work_dir}"
+cd "${work_dir}"
+
+# Equinix Metal ARM server are not yet hourly available in the default `SV` metro
+equinixmetal_metro_var="EQUINIXMETAL_${arch}_METRO"
+equinixmetal_metro="${!equinixmetal_metro_var}"
+
+EQUINIXMETAL_INSTANCE_TYPE_VAR="EQUINIXMETAL_${arch}_INSTANCE_TYPE"
+EQUINIXMETAL_INSTANCE_TYPE="${!EQUINIXMETAL_INSTANCE_TYPE_VAR}"
+MORE_INSTANCE_TYPES_VAR="EQUINIXMETAL_${arch}_MORE_INSTANCE_TYPES"
+MORE_INSTANCE_TYPES=( ${!MORE_INSTANCE_TYPES_VAR} )
+
+# We override `EQUINIXMETAL_PARALLEL`, because running kola with EQUINIXMETAL_PARALLEL >= 4 causes the
+# tests to provision >= 12 ARM servers at the same time. As the DA metro does not
+# have that many free ARM servers, the whole tests will fail.
+# In addition, we override `timeout` to 10 hours, because it takes more than 8 hours
+# to run all tests only with 2 tests in parallel.
+[[ "${arch}" == "arm64" ]] && timeout=15h || timeout=8h
+
+BASE_URL="http://${BUILDCACHE_SERVER}/images/${arch}/${vernum}"
+
+run_equinix_metal_kola_test() {
+    local instance_type="${1}"
+    local instance_tapfile="${2}"
+
+    timeout --signal=SIGQUIT "${timeout}" \
+        kola run \
+          --board="${arch}-usr" \
+          --basename="ci-${vernum/+/-}" \
+          --platform=equinixmetal \
+          --tapfile="${instance_tapfile}" \
+          --parallel="${EQUINIXMETAL_PARALLEL}" \
+          --torcx-manifest=../torcx_manifest.json \
+          --equinixmetal-image-url="${BASE_URL}/${EQUINIXMETAL_IMAGE_NAME}" \
+          --equinixmetal-installer-image-kernel-url="${BASE_URL}/${PXE_KERNEL_NAME}" \
+          --equinixmetal-installer-image-cpio-url="${BASE_URL}/${PXE_IMAGE_NAME}" \
+          --equinixmetal-metro="${equinixmetal_metro}" \
+          --equinixmetal-plan="${instance_type}" \
+          --equinixmetal-project="${EQUINIXMETAL_PROJECT}" \
+          --equinixmetal-storage-url="${EQUINIXMETAL_STORAGE_URL}" \
+          --gce-json-key=<(set +x; echo "${GCP_JSON_KEY}" | base64 --decode) \
+          --equinixmetal-api-key="${EQUINIXMETAL_KEY}" \
+          "${@}"
+
+    # compare the tested instance with the default instance type.
+    if [[ "${instance_type}" != "${EQUINIXMETAL_INSTANCE_TYPE}" ]]; then
+        sed --in-place "s/cl\.internet/${instance_type}\.cl\.internet/" "${instance_tapfile}"
+    fi
+}
+
+cl_internet_included="$(kola list --platform=equinixmetal --filter "${@}" | { grep cl.internet || : ; } )"
+
+# in case of rerun, we need to convert <instance-type>.cl.internet
+# to regular cl.internet tests on the correct instance type.
+instance_types=()
+for t in "${@}"; do
+    if [[ "${t}" =~ ".cl.internet" ]]; then
+        instance_types+=( "${t/\.cl\.internet/}" )
+        # cl_internet needs to run.
+        cl_internet_included="yes"
+        # TODO: replace <instance-type>.cl.internet with cl.internet in ${@}
+    fi
+done
+
```

Comment:
```suggestion

# Remove <instance-type>.cl.internet in ${@}
set -o noglob
set -- $(echo "$*" | sed 's/[^[:space:]]*\.cl\.internet//g')
set +o noglob
```
We just need to remove the prefixed cl.internet names and don't need to insert cl.internet